Azure.ResourceManager.Core.Tests/LocationTests.cs

```diff
@@ -1,110 +1,173 @@
+            Location loc1 = left;
+            Location loc2 = right;
+            Assert.AreEqual(expected, loc1.Equals(loc2));
+
+            if (expected)
+            {
+                Assert.AreEqual(0, loc1.CompareTo(loc2));
+            }
+            else
+            {
+                Assert.AreNotEqual(0, loc1.CompareTo(loc2));
+            }
         }
 
-        [Test]
-        public void CanCompareToStringGreaterThan()
+        [TestCase(true, "West Us", "West Us")]
+        [TestCase(true, "West Us", "WestUs")]
+        [TestCase(false, "West Us", "WestUs2")]
+        [TestCase(false, "West Us", "")]
+        [TestCase(false, "West Us", "!#()@(#@")]
+        [TestCase(false, "West Us", "W3$t U$")]
+        [TestCase(false, "West Us", null)]
+        public void EqualsToString(bool expected, string left, string right)
         {
-            Location location1 = new Location("South US");
-            Assert.AreEqual(1, location1.CompareTo("East US"));
+            Location location = left;
+            Assert.AreEqual(expected, location.Equals(right));
+
+            if (expected)
+            {
+                Assert.AreEqual(0, location.CompareTo(right));
+            }
+            else
+            {
+                Assert.AreNotEqual(0, location.CompareTo(right));
+            }
         }
 
-        [Test]
-        public void CanCompareToLocationLessThan()
+        [TestCase("", "")]
+        [TestCase("West US", "West US")]
+        [TestCase("west-us", "West US")]
+        [TestCase("westus2", "West US 2")]
+        [TestCase("private-cloud", "Private Cloud")]
+        public void CanParseToString(string name, string expected)
         {
-            Location location1 = new Location("South US");
-            Location location3 = new Location("West US");
-
-            Assert.AreEqual(-1, location1.CompareTo(location3));
+            Location location1 = name;
+            Assert.AreEqual(expected, location1.ToString());
         }
 
-        [Test]
-        public void CanCompareToStringLessThan()
+        [TestCase("West US", "West US", 0)]
+        [TestCase("West US", "west-us", 0)]
+        [TestCase("West US", "westus", 0)]
+        [TestCase("Central Europe", "Central Europe", 0)]
+        [TestCase("Central Europe", "central-europe", 0)]
+        [TestCase("Central Europe", "centraleurope", 0)]
+        [TestCase("South US", "East US", 1)]
+        [TestCase("South US", "east-us", 1)]
+        [TestCase("South US", "West US", -1)]
+        [TestCase("South US", "west-us", -1)]
+        [TestCase("South US", null, 1)]
+        public void CompareToObject(string left, string right, int expected)
         {
-            Location location1 = new Location("South US");
-            Assert.AreEqual(-1, location1.CompareTo("West US"));
+            Location location1 = left;
+            Location location2 = right;
+            Assert.AreEqual(expected, location1.CompareTo(location2));
```

Comment:
Add the reverse check like you do with CompareToString

            if (right != null)
            {
                location1 = right;
                Assert.AreEqual(expected * -1, location1.CompareTo(left));
            }